gridstatus/isone_api/isone_api.py

```diff
@@ -0,0 +1,381 @@
+
+        Args:
+            location_id (int): The ID of the location to retrieve.
+
+        Returns:
+            pandas.DataFrame: A DataFrame containing the location information.
+        """
+        url = f"{BASE_URL}/locations/{location_id}"
+        response = self.make_api_call(url)
+
+        if "Location" not in response:
+            raise NoDataFoundException(f"No data found for location ID: {location_id}")
+
+        location = response["Location"]
+        df = pd.DataFrame([location])
+        return df
+
+    def get_locations_all(self) -> pd.DataFrame:
+        """
+        Get detailed information for all locations.
+
+        Returns:
+            pandas.DataFrame: A DataFrame containing detailed information for all locations.
+        """
+        url = f"{BASE_URL}/locations/all"
+        response = self.make_api_call(url)
+
+        if "Locations" not in response or "Location" not in response["Locations"]:
+            raise NoDataFoundException("No location data found.")
+
+        locations = response["Locations"]["Location"]
+        df = pd.DataFrame(locations)
+        return df
+
+    def _handle_demand(
+        self,
+        df: pd.DataFrame,
+        interval_minutes: int = 60,
+    ) -> pd.DataFrame:
+        """
+        Process demand DataFrame: convert types, rename columns, and add Interval End.
+
+        Args:
+            df (pd.DataFrame): Input DataFrame with demand data.
+            interval_minutes (int): Duration of each interval in minutes. Default is 60.
+
+        Returns:
+            pd.DataFrame: Processed DataFrame.
+        """
+        df["Interval Start"] = pd.to_datetime(df["BeginDate"])
+        df["Interval End"] = df["Interval Start"] + pd.Timedelta(
+            minutes=interval_minutes,
+        )
+        df["Load"] = pd.to_numeric(df["Load"], errors="coerce")
+        df["Location Id"] = pd.to_numeric(df["LocId"], errors="coerce")
+        return df[["Interval Start", "Interval End", "Location", "Location Id", "Load"]]
+
+    @support_date_range("DAY_START")
+    def get_realtime_hourly_demand(
+        self,
+        date,
+        end_date: str | None = None,
+        locations: list[str] = None,
+    ) -> pd.DataFrame:
+        """
+        Get the real-time hourly demand data for specified locations and date range.
+
+        Args:
+            date (str): The start date for the data request. Use "latest" for most recent data.
+            end_date (str | None): The end date for the data request. Only used if date is not "latest".
+            locations (list[str], optional): List of specific location names to request data for.
+                                             If None, data for all locations will be retrieved.
+
+        Returns:
+            pandas.DataFrame: A DataFrame containing the real-time hourly demand data for all requested locations.
+        """
+        all_data = []
+
+        match (date, locations):
```

Comment:
Interesting approach with the match case statement. It's concise; my only concern is it's not consistent with the rest of the codebase.

Reply:
It's kind of the endpoints upstream that cause lead me to this approach, since they offer the endpoints like

```
/dayaheadhourlydemand/current # this gets all core location latest, seems most useful
/dayaheadhourlydemand/current/location/{locationId}] # this is the above but requires an explicit location
/dayaheadhourlydemand/day/{day}/location/{locationId} # this is for historical, also requires explicit location
```

Conceivably I could go down to just two cases and use the latter two endpoints, since we would emulate that first endpoint but just use the explicit locations and loop through them ourselves. Essentially treat the first two match cases the same and consolidate. But that first endpoint seems the most useful for what we are probably wanting to do with the data so it felt strange to not use it.

Reply:
Ah yeah, either works. Once I find myself putting `and` and conditional statements in my ifs

```
if date="latest" and not locations:
``` 
or start nesting 
```
if date = latest: 
          if locations:
``` 
I tend to switch to match-case, but this one could probably go either way for sure.

Reply:
im fine with match-case. it's clear what's going on, even if we don't typically use it.